app/models/delete_queue_job.rb

```diff
@@ -90,6 +90,10 @@ def perform
         nil
       end
 
+      # remove any embedded documents from study_file before queueing for deletion as validation errors can occur
+      unset_embedded_documents(object)
+      object.reload
+
```

Comment:
Rather than hunt down possible validation errors and remove them in advance (and there might be other ways validation errors can sneak into our files besides embedded docs), I think it will be cleaner to change lines 104 to use a pattern like 

```
object.assign_attributes({ queued_for_deletion: true, ... })
 object.save!(validate: false)
```

Although ordinarily we should never skip validations, here that's expressly what we want "I don't care if this document is valid or not, since I'm deleting it anyway"

Reply:
That's a fair point.

Reply:
8c390fb